src/cgame/cg_main.cpp

```diff
@@ -1253,9 +1318,15 @@ void CG_Init( int serverMessageNum, int clientNum, const glconfig_t& gl, const G
 
 	CG_SetMapNameFromServerinfo();
 
+	/* The cgs.mapname variable must already be set at this point. */
+	SetMapInfoFromArena( cgs.mapname );
+
 	// load the new map
 	trap_CM_LoadMap(cgs.mapname);
 
+	/* The map must already be loaded at this point. */
```

Comment:
By the "map being loaded" what you mean is that configstrings must be received from the server and parsed?

Reply:
The code fallbacks on a string from the bsp itself.

Reply:
And the way you're getting this string is by having the *server* read it from the BSP and the client read it from the network.

Reply:
That's possible. Do we have a better way to do this?

Reply:
I just noticed something… if what you say is right, this string was read from the network on every frame when displaying the scoreboard, as this code was previously run when rendering `<levelname>` tag.

The code was doing this:

```sh
static void CG_Rocket_DrawLevelName()
{
	Rocket_SetInnerRML( CG_ConfigString( CS_MESSAGE ), RP_QUAKE );
}
```

Now it does (reusing a global variable set at map load):

```sh
static void CG_Rocket_DrawLevelName()
{

	Rocket_SetInnerRML( cg.mapLongName.c_str(), RP_QUAKE );

}
```

I wonder if such thing could be a cause of:

- https://github.com/Unvanquished/Unvanquished/issues/2043

It would have not be enough or the only one cause as it is known that issue can happen when connecting (so, when no scoreboard is displayed).

Reply:
No it is not doing something on the network every frame. The server sends config strings at the beginning and when one of them changes and the client stores the latest values.